integration/aggregate_documents_compat_test.go

```diff
@@ -612,133 +606,73 @@ func TestAggregateCompatGroup(t *testing.T) {
 	testAggregateStagesCompat(t, testCases)
 }
 
-func TestAggregateCompatGroupDotNotation(t *testing.T) {
+func TestAggregateCompatGroupExpressionDottedFields(t *testing.T) {
 	t.Parallel()
 
-	// Providers Composites, ArrayAndDocuments and Mixed
+	// TODO Use all providers after fixing $sort problem:  https://github.com/FerretDB/FerretDB/issues/2276.
+	//
+	// Currently, providers Composites, DocumentsDeeplyNested and Mixed
 	// cannot be used due to sorting difference.
 	// FerretDB always sorts empty array is less than null.
 	// In compat, for `.sort()` an empty array is less than null.
 	// In compat, for aggregation `$sort` null is less than an empty array.
-	// https://github.com/FerretDB/FerretDB/issues/2276
-
-	providers := []shareddata.Provider{
-		shareddata.Scalars,
-
-		shareddata.Doubles,
-		shareddata.OverflowVergeDoubles,
-		shareddata.SmallDoubles,
-		shareddata.Strings,
-		shareddata.Binaries,
-		shareddata.ObjectIDs,
-		shareddata.Bools,
-		shareddata.DateTimes,
-		shareddata.Nulls,
-		shareddata.Regexes,
-		shareddata.Int32s,
-		shareddata.Timestamps,
-		shareddata.Int64s,
-		shareddata.Unsets,
-		shareddata.ObjectIDKeys,
-
-		// shareddata.Composites,
-		shareddata.PostgresEdgeCases,
-
-		shareddata.DocumentsDoubles,
-		shareddata.DocumentsStrings,
-		shareddata.DocumentsDocuments,
-
-		shareddata.ArrayStrings,
-		shareddata.ArrayDoubles,
-		shareddata.ArrayInt32s,
-		shareddata.ArrayRegexes,
-		shareddata.ArrayDocuments,
-
-		// shareddata.Mixed,
-		// shareddata.ArrayAndDocuments,
-	}
+	providers := shareddata.AllProviders().Remove("Mixed", "Composites", "DocumentsDeeplyNested")
 
 	testCases := map[string]aggregateStagesCompatTestCase{
-		"DocDotNotation": {
+		"NestedInDocument": {
 			pipeline: bson.A{bson.D{{"$group", bson.D{
 				{"_id", "$v.foo"},
 			}}}},
 		},
-		"ArrayDotNotation": {
+		"DeeplyNested": { // Expect non-empty results for DocumentsDeeplyNested provider
+			pipeline: bson.A{bson.D{{"$group", bson.D{
+				{"_id", "$v.a.b.c"},
+			}}}},
+		},
+		"ArrayIndex": {
 			pipeline: bson.A{bson.D{{"$group", bson.D{
 				{"_id", "$v.0"},
 			}}}},
 		},
-		"ArrayDocDotNotation": {
+		"NestedInArray": {
 			pipeline: bson.A{bson.D{{"$group", bson.D{
 				{"_id", "$v.0.foo"},
 			}}}},
 		},
-		"NestedDotNotation": {
+		"NonExistentParent": {
 			pipeline: bson.A{bson.D{{"$group", bson.D{
-				{"_id", "$v.0.foo.0.bar"},
```

Comment:
This example doesn't make a lot of sense because aggregation expressions don't work with array elements notation.
Compat example:

```
db.ArrayDocuments.aggregate({"$group":{"_id":"$v.0.foo.0.bar"}}).sort({"_id": 1})
[ { _id: [] } ]
```

This is already covered by `$v.0.foo` test case.